src/chunk_dispatch_plan.c

```diff
@@ -35,132 +35,75 @@ static CustomScanMethods chunk_dispatch_plan_methods = {
 	.CreateCustomScanState = create_chunk_dispatch_state,
 };
 
-/*
- * Build a target list for a CustomScan node.
- *
- * The CustomScan node needs to provide a modified target list in case of
- * subplan nodes that have target lists with expressions that involve, e.g.,
- * aggregates. Such target lists need a matching parent node type (e.g., a
- * target list with an AggDef needs a Agg node parent). If we simply use the
- * subplan's target list on a CustomScan parent node, there might be a mismatch.
- *
- * The code here is similar to ExecCheckPlanOutput() in nodeModifyTable.c in the
- * PostgreSQL source code, but here we construct a modified target list for the
- * CustomScan node instead of simply just checking validity.
- *
- */
-static List *
-build_customscan_targetlist(Relation rel, List *targetlist)
-{
-	TupleDesc	tupdesc = RelationGetDescr(rel);
-	List	   *new_targetlist = NIL;
-	ListCell   *lc;
-	int			attno = 0;
-
-	foreach(lc, targetlist)
-	{
-		TargetEntry *te = lfirst(lc);
-		Form_pg_attribute attr;
-		Expr	   *expr;
-
-		/* Ignore junk items */
-		if (te->resjunk)
-			continue;
-
-		if (attno >= tupdesc->natts)
-			ereport(ERROR,
-					(errcode(ERRCODE_DATATYPE_MISMATCH),
-					 errmsg("table row type and query-specified row type do not match"),
-					 errdetail("Query has too many columns.")));
-
-		attr = TupleDescAttr(tupdesc, attno);
-		attno++;
-
-		if (attr->attisdropped)
-			expr = &makeConst(INT4OID,
-							  -1,
-							  InvalidOid,
-							  sizeof(int32),
-							  (Datum) 0,
-							  true,
-							  true)->xpr;
-		else
-
-			/*
-			 * According to nodes/primnodes.h, the INDEX_VAR Var type is
-			 * abused in CustomScan nodes to reference custom scan tuple
-			 * types.
-			 */
-			expr = &makeVar(INDEX_VAR,
-							attno,
-							exprType((Node *) te->expr),
-							exprTypmod((Node *) te->expr),
-							exprCollation((Node *) te->expr),
-							0)->xpr;
-
-		new_targetlist = lappend(new_targetlist,
-								 makeTargetEntry(expr,
-												 attno,
-												 NULL,
-												 te->resjunk));
-	}
-
-	if (attno != tupdesc->natts)
-		ereport(ERROR,
-				(errcode(ERRCODE_DATATYPE_MISMATCH),
-				 errmsg("table row type and query-specified row type do not match"),
-				 errdetail("Query has too few columns.")));
-
-	return new_targetlist;
-}
-
 /* Create a chunk dispatch plan node in the form of a CustomScan node. The
  * purpose of this plan node is to dispatch (route) tuples to the correct chunk
  * in a hypertable.
  *
  * Note that CustomScan nodes cannot be extended (by struct embedding) because
- * they might be copied, therefore we pass hypertable_relid in the
- * custom_private field.
+ * they might be copied, therefore we pass any extra info as a ChunkDispatchInfo
```

Comment:
This seems to be rebase error since ChunkDispatchInfo was removed and the code stores oid directly in custom_private